src/schema/schema_create.c

```diff
@@ -980,6 +980,23 @@ __schema_create(WT_SESSION_IMPL *session, const char *uri, const char *config)
         WT_RET_MSG(session, ENOTSUP,
           "%s: import is only supported for 'file' and 'table' data sources", uri);
 
+    /*
+     * If tiered storage is configured at the connection level and the user has not configured
+     * tiered_storage.name to be none, then the object being created is a tiered object.
+     */
+    is_tiered = S2C(session)->bstorage != NULL &&
+      !((ret = __wt_config_getones(session, config, "tiered_storage.name", &cval)) == 0 &&
+        cval.len != 0 && WT_STRING_MATCH("none", cval.str, cval.len));
```

Comment:
The logic above will detect it as a tiered case if `__wt_config_getones` fails or `cval.len` is `0`. This is not what we want. We need to rework the logic. To me, the best way is to extract the code that gets tiered storage name out of the evaluation. It will simplify the expression.

Reply:
Or is it a tiered case if there's no `tiered_storage.name` or it's empty?

Reply:
If tiered is configured on the connection (i.e. bstorage != NULL) then every table is tiered by default. The only way to have a local-only table in a tiered connection is to set `tiered_storage=(name=none)`. So if the `name` is empty on a table create it defaults to the name in `conn->bstorage`.